pkg/collector/clock.go

```diff
@@ -70,6 +94,9 @@ func (t *fakeTimer) Stop() bool {
 }
 
 func (t *fakeTimer) Reset(d time.Duration) bool {
+	if d <= 0 {
```

Comment:
If I accept `d < 0`, I have to potentially call the step function here, which would just add complexity. This can be supported at a later time if needed.